spec/ios_download_strings_files_from_glotpress_spec.rb

```diff
@@ -0,0 +1,181 @@
+require 'spec_helper'
+require 'tmpdir'
+
+describe Fastlane::Actions::IosDownloadStringsFilesFromGlotpressAction do
+  let(:test_data_dir) { File.join(File.dirname(__FILE__), 'test-data', 'translations', 'ios_generate_strings_file_from_code') }
+  let(:gp_fake_url) { 'https://stub.glotpress.com/rspec-fake-project' }
+  let(:locales_subset) { { 'fr-FR': 'fr', 'zh-cn': 'zh-Hans' } }
+
+  def gp_stub(locale:, query:)
+    stub_request(:get, "#{gp_fake_url}/#{locale}/default/export-translations").with(query: query)
+  end
+
+  describe 'downloading export files from GlotPress' do
+    def test_gp_download(filters:, tablename:, expected_gp_params:)
+      Dir.mktmpdir('a8c-release-toolkit-tests-') do |tmp_dir|
+        # Arrange
+        stub_fr = gp_stub(locale: 'fr-FR', query: expected_gp_params).to_return(body: '"key" = "fr-copy";')
+        stub_zh = gp_stub(locale: 'zh-cn', query: expected_gp_params).to_return(body: '"key" = "zh-copy";')
+
+        # Act
+        run_described_fastlane_action({
+          project_url: gp_fake_url,
+          locales: locales_subset,
+          download_dir: tmp_dir,
+          table_basename: tablename,
+          filters: filters
+        }.compact)
```

Comment:
This `.compact` ensures that if we pass a `nil` value for `tablename` or `filters`, then the corresponding keys / parameters passed when calling the action (i.e. `table_basename: …` and `filters: …` won't be included in the call site; allowing them to use the default value of their `ConfigItem` instead, just as intended (as opposed to providing the parameters but passing them a `nil` value which would overwrite the default)

Reply:
Nice. What do you think of adding the comment in the code? I can see myself scratching my head for a minute when looking at this code in the future.

Reply:
Good point, will add one, thanks for the suggestion  👍

Reply:
Added via ba0fdc6d9ee7671b3888f4c838c323644e11caca 🙂